src/test/java/seedu/partyplanet/logic/commands/EEditCommandTest.java

```diff
@@ -0,0 +1,182 @@
+package seedu.partyplanet.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.partyplanet.logic.commands.CommandTestUtil.DESC_CNY;
+import static seedu.partyplanet.logic.commands.CommandTestUtil.DESC_EASTER;
+import static seedu.partyplanet.logic.commands.CommandTestUtil.VALID_NAME_EASTER;
+import static seedu.partyplanet.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.partyplanet.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.partyplanet.logic.commands.CommandTestUtil.showEventAtIndex;
+import static seedu.partyplanet.testutil.TypicalEvents.getTypicalEventBook;
+import static seedu.partyplanet.testutil.TypicalIndexes.INDEX_FIRST_EVENT;
+import static seedu.partyplanet.testutil.TypicalIndexes.INDEX_SECOND_EVENT;
+import static seedu.partyplanet.testutil.TypicalPersons.getTypicalAddressBook;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.partyplanet.commons.core.Messages;
+import seedu.partyplanet.commons.core.index.Index;
+import seedu.partyplanet.logic.commands.EEditCommand.EditEventDescriptor;
+import seedu.partyplanet.model.AddressBook;
+import seedu.partyplanet.model.EventBook;
+import seedu.partyplanet.model.Model;
+import seedu.partyplanet.model.ModelManager;
+import seedu.partyplanet.model.UserPrefs;
+import seedu.partyplanet.model.event.Event;
+import seedu.partyplanet.testutil.EditEventDescriptorBuilder;
+import seedu.partyplanet.testutil.EventBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for EEditCommand.
+ */
+public class EEditCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), getTypicalEventBook(), new UserPrefs());
+
+    @Test
+    public void execute_allFieldsSpecifiedUnfilteredList_success() {
+        Event editedEvent = new EventBuilder().build();
+        EditEventDescriptor descriptor = new EditEventDescriptorBuilder(editedEvent).build();
+        EEditCommand eeditCommand = new EEditCommand(INDEX_FIRST_EVENT, descriptor);
+
+        String expectedMessage = String.format(EEditCommand.MESSAGE_EDIT_EVENT_SUCCESS, editedEvent);
+
+        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()),
+                new EventBook(model.getEventBook()), new UserPrefs());
+        expectedModel.setEvent(model.getFilteredEventList().get(0), editedEvent);
```

Comment:
To avoid "magic numbers"

```suggestion
        expectedModel.setEvent(model.getFilteredEventList().get(INDEX_FIRST_EVENT.getZeroBased()), editedEvent);
```